Makefile

```diff
@@ -254,20 +269,25 @@ LOCALBIN ?= $(shell pwd)/bin
 $(LOCALBIN):
 	mkdir -p $(LOCALBIN)
 
-# Download controller-gen locally if necessary
+# Download controller-gen locally if necessary. If wrong version is installed, it will be overwritten.
 CONTROLLER_GEN = $(LOCALBIN)/controller-gen
 .PHONY: controller-gen
 controller-gen: $(CONTROLLER_GEN)
 $(CONTROLLER_GEN): $(LOCALBIN)
-	test -s $(CONTROLLER_GEN) || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_GEN_VERSION)
+	test -s $(CONTROLLER_GEN) && $(CONTROLLER_GEN) --version | grep -q $(CONTROLLER_TOOLS_VERSION) || \
```

Comment:
I think what's happening is that the `$(LOCALBIN)` directory has the same modification time as the most recently updated binary within it. Because `$(LOCALBIN)` is used as a prerequisite for targets like controller-gen, Make sees that the prerequisite is not newer than the target, so there's nothing to do.

One fix could be to use a phony target for `$(LOCALBIN)` and not use the actual directory as a prerequisite anywhere. This would ensure the directory exists, but not considering its modification time when running dependent targets.

Something like this:
```patch
diff --git a/Makefile b/Makefile
index 4ae65c9..788c668 100644
--- a/Makefile
+++ b/Makefile
@@ -266,14 +266,15 @@ check_cert_manager:
 
 # Location to install dependencies
 LOCALBIN ?= $(shell pwd)/bin
-$(LOCALBIN):
-	mkdir -p $(LOCALBIN)
+.PHONY: local-bin
+local-bin:
+	test -d $(LOCALBIN) || mkdir -p $(LOCALBIN)
 
 # Download controller-gen locally if necessary. If wrong version is installed, it will be overwritten.
 CONTROLLER_GEN = $(LOCALBIN)/controller-gen
 .PHONY: controller-gen
 controller-gen: $(CONTROLLER_GEN)
-$(CONTROLLER_GEN): $(LOCALBIN)
+$(CONTROLLER_GEN): local-bin
 	test -s $(CONTROLLER_GEN) && $(CONTROLLER_GEN) --version | grep -q $(CONTROLLER_TOOLS_VERSION) || \
 	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@v$(CONTROLLER_TOOLS_VERSION)
 
@@ -282,7 +283,7 @@ KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/k
 KUSTOMIZE = $(LOCALBIN)/kustomize
 .PHONY: kustomize
 kustomize: $(KUSTOMIZE)
-$(KUSTOMIZE): $(LOCALBIN)
+$(KUSTOMIZE): local-bin
 	@if test -x $(LOCALBIN)/kustomize && ! $(LOCALBIN)/kustomize version | grep -q $(KUSTOMIZE_VERSION); then \
 		echo "$(LOCALBIN)/kustomize version is not expected $(KUSTOMIZE_VERSION). Removing it before installing."; \
 		rm -rf $(LOCALBIN)/kustomize; \
@@ -293,21 +294,21 @@ $(KUSTOMIZE): $(LOCALBIN)
 ADDLICENSE = $(LOCALBIN)/addlicense
 .PHONY: addlicense
 addlicense: $(ADDLICENSE)
-$(ADDLICENSE): $(LOCALBIN)
+$(ADDLICENSE): local-bin
 	test -s $(ADDLICENSE) || GOBIN=$(LOCALBIN) go install github.com/google/addlicense@v$(ADDLICENSE_VERSION)
 
 # Download setup-envtest locally if necessary
 ENVTEST = $(LOCALBIN)/setup-envtest
 .PHONY: setup-envtest
 setup-envtest: $(ENVTEST)
-$(ENVTEST): $(LOCALBIN)
+$(ENVTEST): local-bin
 	test -s $(ENVTEST) || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
 
 # Download opm locally if necessary
 OPM = $(LOCALBIN)/opm
 .PHONY: opm
 opm: $(OPM)
-$(OPM): $(LOCALBIN)
+$(OPM): local-bin
 	test -s $(OPM) || \
 	{ \
 	set -e ;\
```

Reply:
The `test -d` can be removed since we already use `mkdir -p`.